features/dd-sdk-android-rum/src/main/kotlin/com/datadog/android/telemetry/internal/TelemetryEventHandler.kt

```diff
@@ -198,12 +201,22 @@ internal class TelemetryEventHandler(
         )
     }
 
+    @Suppress("LongMethod")
     private fun createConfigurationEvent(
         datadogContext: DatadogContext,
         timestamp: Long,
         coreConfiguration: TelemetryCoreConfiguration
     ): TelemetryConfigurationEvent {
         val traceFeature = sdkCore.getFeature(Feature.TRACING_FEATURE_NAME)
+        val sessionReplayFeatureContext =
+            sdkCore.getFeatureContext(Feature.SESSION_REPLAY_FEATURE_NAME)
+        val sessionReplaySampleRate = sessionReplayFeatureContext[SESSION_REPLAY_SAMPLE_RATE_KEY]
+            ?.let { it as? Long }
```

Comment:
why do we need `let`? we should be able to write as `sessionReplayFeatureContext[SESSION_REPLAY_SAMPLE_RATE_KEY] as? Long` and I guess the result should be the same? same for the lines below.